### PR TITLE
[WIP ] #151 Edit events

### PR DESCRIPTION
**Pull request by abdellani**
#### Issue addressed

https://github.com/AgileVentures/WebsiteOne-FE/issues/151

### CODE REVIEW

**Comment by abdellani** [comment]
@mattwr18 
At this level, users have access to the edit event form with data preloaded. Now, I wonder how to implement the change handlers for the form? is it better to do the changes at the component's state  or do I need to use actions/store  ?

**Comment by mattwr18** [comment]
hey @abdellani there are 4 jest tests that are failing on the build, are they failing in the same way locally?

**Comment by abdellani** [comment]
@mattwr18 
Is `/api/v1/events/${slug}` supposed to send data structured in the same way like in [cypress/fixtures/event.json](https://github.com/AgileVentures/WebsiteOne-FE/blob/develop/cypress/fixtures/event.json)  ? 
I'm asking because I faced a problem while writing cypress test. When I try to edit the event,  [src/actions/getEventInfoAction.js](https://github.com/AgileVentures/WebsiteOne-FE/blob/develop/src/actions/getEventInfoAction.js), The summary page doesn't show all the details, like the name for example.

Now, I'm using this [fixture](https://gist.github.com/abdellani/92d4967ebc7ae02888e339235521c8ab) for fetching event's info.

File: src/actions/updateEventAction.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      repeat_ends_on: endDate,
+      token: headers
+    },
+    headers: {
+      Authorization: headers,
+      Accept: 'application/json'
+    }
+  })
+    .then(response => {
+      history.push(`/events/${response.data.event.slug}`)
```
Comment (mattwr18) [inline src/actions/updateEventAction.js:55]: We were talking about this is in the meeting @abdellani 
@joaopapereira has requested that we extract this concern from the action into either the reducer or the container...
so far we have other examples, i.e `Login.js`, where it is the concern of the container
Maybe, though we just want to open a ticket and do them all at the same time?

File: src/containers/EditEventPage.js [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+  componentDidMount () {
+    const path = this.props.location.pathname
+    this.props.setLastLocation(path)
+    if (!this.props.cookies.get(process.env.SESSION || 'WebsiteOne_session') && !this.props.loggedInUser.data) {
+      this.props.history.push({ pathname: '/login' })
+    }
+    if (!this.props.projects.length) {
+      this.props.fetchActiveProjects()
+      this.props.fetchEventInfo(this.props.match.params.slug)
```
Comment (mattwr18) [inline src/containers/EditEventPage.js]: it seems a bit strange to have a `fetchEventInfo` nested inside an if block checking about the length of the projects array, doesn't it? I think I left it there in our pairing session?

File: src/containers/EditEventPage.js [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+//   }
+//   ,
+//   (dispatch, ownProps) => {
+//     let { slug } = ownProps.match.params
+//     return {
+//       fetchEventInfo: () => dispatch(fetchEventInfo(slug)),
+//       updateEvent
+//     }
+//   }
+// )(EditEventPage)
```
Comment (mattwr18) [inline src/containers/EditEventPage.js]: please remove the comments :smile_cat:

**Review by mattwr18 — CHANGES_REQUESTED** [review]

**Comment by mattwr18** [comment]
this is great @abdellani, I think we can get this merged soon... I'll pull down the latest changes and check it out.. did you see why the test coverage dropped?

File: src/containers/EditEventPage.js [hunk 4]
```diff
@@ -101,7 +99,7 @@ export class EditEventPage extends Component {
         ? <CustomRingLoader />
         : <Fragment>
           <Header as='h1' textAlign='center'>
-          Creating a new Event
+            Creating a new Event
```
Comment (mattwr18) [inline src/containers/EditEventPage.js]: should this be `Editing Event`, or something similar @abdellani?

**Review by mattwr18 — CHANGES_REQUESTED** [review]

**Comment by aonomike** [comment]
I think the test coverage dropped since there are no tests for `updateEventsAction.js`. Maybe an update to that would improve the coverage @abdellani @mattwr18  ?

**Pull request closed without merging**